src/openassetio-python/package/openassetio/managerApi/ManagerInterface.py

```diff
@@ -479,85 +479,104 @@ def finalizedEntityVersion(self, entityRefs, context, hostSession, overrideVersi
+    def getWithRelationship(
+        self, relationshipTraitsData, toEntityReferences, context, hostSession, resultTraitSet=None
     ):
         """
-        Returns related entity references, based on a relationship
-        defined by a set of traits and their properties.
+        Return entity references that are related to the input
+        references by the relationship defined by a set of traits and
+        their properties.
 
         This is an essential function in this API - as it is widely used
-        to query organisational hierarchy, etc...
+        to query other entities or organisational structure.
 
-        There are three possible conventions for calling this function,
-        to allow for batch optimisations in the implementation and
-        prevent excessive query times with high-latency services.
+        @note The order of entities in the inner lists of matching
+        references should not be considered meaningful, but the outer
+        list should match the input entity reference order.
 
-         - a)  A single entity reference, a list of relationships.
-         - b)  A list of entity references and a single relationship.
-         - c)  Equal length lists of references and relationship.
+        If any relationship definition is unknown, then an empty list
+        should be returned for that entity, and no errors raised. The
+        default implementation returns an empty list for all
+        relationships.
 
-        In all cases, the return value is a list of lists, for example:
+        @param relationshipTraitsData @fqref{TraitsData} "TraitsData"
+        The traits of the relationship to query.
 
-            a)  getRelatedReferences([ r1 ], [ td1, td2, td3 ])
+        @param toEntityReferences `List[` @fqref{EntityReference} "EntityReference" `]`
+        A list of @ref entity_reference to query the specified
+        relationship for.
 
-            > [ [ r1td1... ], [ r1td2... ], [ r1td3... ] ]
+        @param context Context The calling context.
 
-            b)  getRelatedReferences([ r1, r2, r3 ], [ td1 ])
+        @param hostSession openassetio.managerApi.HostSession The host
+        session that maps to the caller, this should be used for all
+        logging and provides access to the openassetio.managerApi.Host
+        object representing the process that initiated the API session.
 
-            > [ [ r1td1... ], [ r2td1... ], [ r3td1... ] ]
+        @param resultTraitSet `Set[str]` or None, a hint as to what
+        traits the returned entities should have.
+
+        @return `List[List[@fqref{EntityReference} "EntityReference"`]]`
+        A list of references to related entities, for each input
+        reference.
+
+        @note Ensure that your implementation of
+        @fqref{managerApi.ManagerInterface.managementPolicy}
+        "managementPolicy" responds appropriately when queried with
+        relationship trait sets and a read policy to communicate to the
```

Comment:
[Consider][Minor]

```suggestion
        relationship trait sets and has a read policy to communicate to the
```

Reply:
I think the original version is more accurate as its describing two inputs to the query?